src/OVAL/probes/probe/icache.c

```diff
@@ -457,39 +457,26 @@ int probe_icache_nop(probe_icache_t *cache)
         return (0);
 }
 
-#define PROBE_RESULT_MEMCHECK_CTRESHOLD  32768  /* item count */
-#define PROBE_RESULT_MEMCHECK_MINFREEMEM 512    /* MiB */
-#define PROBE_RESULT_MEMCHECK_MAXRATIO   0.8   /* max. memory usage ratio - used/total */
+/* Every n-th call of probe_cobj_memcheck actually checks memory. This is due to performance reasons */
+#define PROBE_RESULT_MEMCHECK_FREQUENCY  100
+#define PROBE_RESULT_MEMCHECK_MINFREEMEM 256    /* MiB */
 
 /**
  * Returns 0 if the memory constraints are not reached. Otherwise, 1 is returned.
  * In case of an error, -1 is returned.
  */
-static int probe_cobj_memcheck(size_t item_cnt)
+static int probe_cobj_memcheck()
 {
-	if (item_cnt > PROBE_RESULT_MEMCHECK_CTRESHOLD) {
-		struct proc_memusage mu_proc;
+	static __thread unsigned long cnt = PROBE_RESULT_MEMCHECK_FREQUENCY;
```

Comment:
It seems to me that we have to use a different way to declare thread-local storage in MSVC. Please see https://msdn.microsoft.com/en-us/library/6yh4a9k1.aspx